src/webots/scene_tree/WbInsertExternProtoDialog.cpp

```diff
@@ -133,7 +144,12 @@ void WbInsertExternProtoDialog::accept() {
   }
 
   // the addition must be declared as EXTERNPROTO so that it is added to the world file when saving
+  if (mPath.startsWith(WbStandardPaths::webotsHomePath()))
+    mPath = mPath.replace(WbStandardPaths::webotsHomePath(), "webots://");
+  if (mPath.startsWith(WbProject::current()->protosPath()))
+    mPath = QDir(WbProject::current()->worldsPath()).relativeFilePath(mPath);
   WbProtoManager::instance()->declareExternProto(mProto, mPath, true);
+  WbLog::info(tr("You need to save the world for the declaration to be taken into consideration."));
```

Comment:
I'm not sure if a user reading this message in the console will understand what declaration it refers to.
Probably it would be better to add it in the text. For example
```suggestion
  WbLog::info(tr("You need to save the world for the EXTERNPROTO declaration to be taken into consideration."));
```